MergeSharpWebAPI/Models/Graph.cs

```
@@ -0,0 +1,177 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MergeSharp;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace MergeSharpWebAPI.Models;
+
+[TypeAntiEntropyProtocol(typeof(Graph))]
+public class GraphMsg : PropagationMessage
+{
+    [JsonInclude]
+    public CanilleraGraphMsg cGraphMsg { get; private set; }
+
+    [JsonInclude]
+    public Dictionary<Guid, VertexInfoMsg> vertexInfoMsgs { get; private set; }
+
+    public GraphMsg()
+    {
+        this.cGraphMsg = new();
+        this.vertexInfoMsgs = new();
+    }
+
+    public GraphMsg(CanilleraGraph cGraph, Dictionary<Guid, VertexInfo> vertexInfo)
+    {
+        this.cGraphMsg = (CanilleraGraphMsg) cGraph.GetLastSynchronizedUpdate();
+        this.vertexInfoMsgs = vertexInfo.ToDictionary(kv => kv.Key, kv => (VertexInfoMsg) kv.Value.GetLastSynchronizedUpdate());
+    }
+
+    public override void Decode(byte[] input)
+    {
+        var json = JsonSerializer.Deserialize<GraphMsg>(input);
+        if (json is null)

```

TODO: this should throw an error